src/circuit/lookup.rs

```diff
@@ -9,7 +9,18 @@ use bellman::{
     ConstraintSystem
 };
 
-// Synthesize the constants for each base pattern.
+/// Synthesize the constants for each base pattern:
+/// Given table `I`, viewed as a function from bit strings to Fr,
+/// we compute in `assignment` the coefficients of the multi-linear polynomial P describing I.
+/// For example, when `usize` is 2, then P(b0,b1)=a0 + a1*b0 + a2*b1 + a3*b0b1,
+/// and P(b0,b1) = I(b), where b is the integer constructed from (b0,b1) in little endian order.
+/// The coeff a_b is defined inductively on  b as a_b=I(b)-sum a_{b'}
+/// running over b' that are "contained" in b; meaning that the ones in the binary representation of b' are
+/// contained in the ones of the binary representation of b.
+/// Under this definition, one can show that P(b) = I(b) - basically all terms except I(b) are
```

Comment:
Don't use "-" as punctuation here! I read this as "minus basically all terms ..." at first and was very confused. :-p

Reply:
Switched to ":"